### PR TITLE
longer retry on any exception

### PR DESCRIPTION
**Pull request by danielwitz**

### CODE REVIEW

File: src/main/java/target/HttpTarget.java [hunk 1]
```diff
@@ -156,7 +156,7 @@ public CompletableFuture<Object> call(final ConsumerRecord<String, String> recor
                 connectionFailureDelayFactor
             )
             .withMaxRetries(Config.CONNECTION_FAILURE_RETRY_POLICY_MAX_RETRIES)
-            .handle(ConnectException.class)
```
Comment (omerlh) [inline src/main/java/target/HttpTarget.java:159]: We probably need to remove the `handle` from the other policy

Comment (danielwitz) [reply src/main/java/target/HttpTarget.java:159]: I think we should keep it. All it does it gives us more retries

File: tests/specs/tests.spec.ts [hunk 2]
```diff
@@ -367,7 +367,7 @@ describe('tests', () => {
         });
     }, 1800000);
 
-    it('consumer should wait after connection error from target', async () => {
+    it.only('consumer should wait after connection error from target', async () => {
```
Comment (guysegal) [inline tests/specs/tests.spec.ts:370]: remove only

**Comment by guysegal** [comment]
change naming

**Comment by danielwitz** [comment]
> change naming

@guysegal done

File: src/main/java/target/HttpTarget.java (hunk 1 above)
Comment (omerlh) [reply src/main/java/target/HttpTarget.java:159]: i am not sure what happens when policies clash so I think we should be careful here

Comment (danielwitz) [reply src/main/java/target/HttpTarget.java:159]: the last one on the list (connectionFailureRetryPolicy) acts first and when it's done the other one (retryPolicy) does

**Pull request merged**